src/hw3/safe_dictionary.py

```diff
@@ -0,0 +1,13 @@
+from contextlib import contextmanager
+from threading import Lock
+
+
+class SafeDictionary:
+    def __init__(self):
+        self._dict = {}
+        self._lock = Lock()
+
+    @contextmanager
```

Comment:
Nice use of context lib, however the homework was to implement context manager from scratch.
This time is okay), next please more attention to HW